File: contest/atcoder/s8pc-2/b/0.cpp
// [25, 32] で RE。
// 当然 AC とはならない素朴な実装。部分点 25 点を得た。
#include <cstdio>

long n;
long numbers[1000001];
int q;
int a;

int main(void)
{
	scanf("%ld%d", &n, &q);
	for (int i = 1; i <= n; i++) {
		numbers[i] = i;
	}
	for (int i = 0; i < q; i++) {
		scanf("%d", &a);
		for (int j = 1; j <= n; j++) {
			if (numbers[j] % a == 0) {
				numbers[j] /= a;
			}
		}
	}
	int answer = 0;
	for (int i = 1; i <= n; i++) {
		if (numbers[i] == 1) {
			answer++;
		}
	}
	printf("%d\n", answer);
	return 0;
}

File: contest/atcoder/s8pc-2/b/0.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
# 間違っている。
n, q = list(map(int, input().split()))
print('n = ' + str(n))
numbers = {}
numbers['range'] = (1, n)
numbers['deads'] = []
numberss = [numbers]
print('q = ' + str(q))
for i in range(q):
    print('i = ' + str(i))
    a = int(input())
    print('a = ' + str(a))
    new_numberss = []
    for numbers in numberss:
        new_numbers = {}
        new_numbers['range'] = (1, numbers['range'][1] // a)
        new_numbers['deads'] = []
        deads = numbers['deads']
        for dead in deads:
            if dead % a == 0:
                new_numbers['deads'].append(dead / a)
            else:
                new_numbers['deads'].append(dead)
        numbers['deads'].append(a)
        new_numberss.append(new_numbers)
    numberss.extend(new_numberss)
    print(numberss)
answer = 0
for numbers in numberss:
    if 1 not in numbers['deads']:
        answer += 1
print(answer)
            
